src/core/simulation.jl

```diff
@@ -0,0 +1,569 @@
+        horizon = get_stage_horizon(get_sequence(sim), stage_name)
+        seq_interval = get_stage_interval(get_sequence(sim), stage_name)
+        if PSY.are_forecasts_contiguous(stage_system)
+            stage_initial_times[stage_number] =
+                PSY.generate_initial_times(stage_system, seq_interval, horizon)
+            if isempty(stage_initial_times[stage_number])
+                throw(IS.ConflictingInputsError("Simulation interval ($seq_interval) and
+                        forecast interval ($interval) definitions are not compatible"))
+            end
+        else
+            stage_initial_times[stage_number] = PSY.get_forecast_initial_times(stage_system)
+            interval = PSY.get_forecasts_interval(stage_system)
+            if interval != seq_interval
+                throw(IS.ConflictingInputsError("Simulation interval ($seq_interval) and
+                        forecast interval ($interval) definitions are not compatible"))
+            end
+            for (ix, element) in enumerate(stage_initial_times[stage_number][1:(end - 1)])
+                if !(element + interval == stage_initial_times[stage_number][ix + 1])
+                    throw(IS.ConflictingInputsError("The sequence of forecasts is invalid"))
+                end
+            end
+        end
+        if !isnothing(sim_ini_time) &&
+           !mapreduce(x -> x == sim_ini_time, |, stage_initial_times[stage_number])
+            throw(IS.ConflictingInputsError("The specified simulation initial_time $sim_ini_time isn't contained in stage $stage_number.
+            Manually provided initial times have to be compatible with the specified interval and horizon in the stages."))
+        end
+        stage_number == 1 && (time_range[1] = stage_initial_times[stage_number][1])
+        (
+            stage_number == k_size &&
+            (time_range[end] = stage_initial_times[stage_number][end])
+        )
+    end
+    sim.internal.date_range = Tuple(time_range)
+
+    if isnothing(get_initial_time(sim))
+        sim.initial_time = stage_initial_times[1][1]
+        @debug("Initial Simulation will be infered from the data.
+               Initial Simulation Time set to $(sim.initial_time)")
+    end
+
+    return stage_initial_times
+end
+
+function _attach_feedforward!(sim::Simulation, stage_name::String)
+    stage = get(sim.stages, stage_name, nothing)
+    feedforward = filter(p -> (p.first[1] == stage_name), sim.sequence.feedforward)
+    for (key, ff) in feedforward
+        # Note: key[1] = Stage name, key[2] = template field name, key[3] = device model key
+        field_dict = getfield(stage.template, key[2])
+        device_model = get(field_dict, key[3], nothing)
+        isnothing(device_model) &&
+        throw(IS.ConflictingInputsError("Device model $(key[3]) not found in stage $(stage_name)"))
+        device_model.feedforward = ff
+    end
+    return
+end
+
+function _check_steps(
```

Comment:
add a check to make sure that SimulationSequence.step_resolution is not longer than the first stage interval. 
The following error is hard to understand:
```
julia> sequence = SimulationSequence(
           step_resolution = Hour(24),
           order = Dict(1 => "UC", 2 => "ED"),
           feedforward_chronologies = Dict(("UC" => "ED") => Synchronize(periods = 12)),
           horizons = Dict("UC" => 18, "ED" => 12),
           intervals = Dict(
               "UC" => (Hour(12), Consecutive()),
                       "ED" => (Hour(1), Consecutive()),
           ),
           feedforward = Dict(
               ("ED", :devices, :Generators) => SemiContinuousFF(
                   binary_from_stage = PSI.ON,
                   affected_variables = [PSI.ACTIVE_POWER],
               ),
           ),
           cache = Dict("UC" => [TimeStatusChange(PSY.ThermalStandard, PSI.ON)]),
           ini_cond_chronology = InterStageChronology(),
       )
Feed Forward Chronology
-----------------------

ED: SemiContinuousFF -> Generators

                     UC--┐ from : On
                         |
┌----┬----┬----┬----┬----┼----┬----┬----┬----┬----┬----┐
|    |    |    |    |    |    |    |    |    |    |    |
|    |    |    |    |    |    |    |    |    |    |    |
└─ED └─ED └─ED └─ED └─ED └─ED └─ED └─ED └─ED └─ED └─ED └─ED (12) to : ["P"]

Initial Condition Chronology
----------------------------

1
|
|
2 --> 2 ... (x12)   


julia> 

julia> f = joinpath(rts_siip_dir,"test_rts")
"/var/folders/27/2jr8c7gn4j72fvrg4qt81zrw8w_711/T/RTS-GMLC-master/RTS_Data/FormattedData/SIIP/test_rts"

julia> file_path = ispath(f) ? f : mkdir(f)
"/var/folders/27/2jr8c7gn4j72fvrg4qt81zrw8w_711/T/RTS-GMLC-master/RTS_Data/FormattedData/SIIP/test_rts"

julia> sim = Simulation(
           name = "aggregation",
           steps = 1,
           stages = stages_definition,
           stages_sequence = sequence,
           simulation_folder = file_path,
           #initial_time = Dates.DateTime("2020-06-05T00:00:00")
           )
Simulation()


julia> build!(sim)
ERROR: AssertionError: length(findall((x->begin
                    #= /Users/cbarrows/Documents/repos/PowerSimulations.jl/src/core/simulation.jl:278 =#
                    x == stage_number
                end), sim.sequence.execution_order)) == execution_counts
Stacktrace:
 [1] _check_steps(::Simulation, ::Dict{Int64,Array{DateTime,1}}) at /Users/cbarrows/Documents/repos/PowerSimulations.jl/src/core/simulation.jl:278
 [2] macro expansion at /Users/cbarrows/Documents/repos/PowerSimulations.jl/src/core/simulation.jl:384 [inlined]
 [3] macro expansion at /Users/cbarrows/.julia/packages/TimerOutputs/7Id5J/src/TimerOutput.jl:214 [inlined]
 [4] #build!#320(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::typeof(build!), ::Simulation) at /Users/cbarrows/Documents/repos/PowerSimulations.jl/src/core/simulation.jl:363
 [5] build!(::Simulation) at /Users/cbarrows/Documents/repos/PowerSimulations.jl/src/core/simulation.jl:362
 [6] top-level scope at REPL[12]:1
```

Reply:
fixed via 3ee2f88